src/runtime/pkg/kata-monitor/cri.go

```diff
@@ -188,11 +188,14 @@ func (km *KataMonitor) getSandboxes(sandboxMap map[string]bool) (map[string]bool
 		}
 
 		// If lowRuntime is empty something changed in containerd/CRI-O or we are dealing with an unknown container engine.
-		// Safest options is to add the POD in the list: we will be able to connect to the shim to retrieve the actual info
-		// only for kata PODs.
+		// Let's check if we have the kata shim socket: this would be evidence that the POD is a kata one.
 		if lowRuntime == "" {
 			monitorLog.WithField("pod", r).Warning("unable to retrieve the runtime type")
-			newMap[pod.Id] = true
+			if checkShimSocketExists(pod.Id) {
```

Comment:
Does newMap[pod.Id] = checkShimSocketExists(pod.Id) work?
No need to re-spin for this, just wondering

Reply:
right, it should be the way to do it!